airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/unstructured_parser.py

```diff
@@ -123,10 +216,12 @@ def _read_file(self, file_handle: IOBase, remote_file: RemoteFile, format: Unstr
         elif filetype == FileType.PPTX:
             elements = unstructured_partition_pptx(file=file)
 
-        return self._render_markdown(elements)
+        return self._render_markdown([element.to_dict() for element in elements])
 
-    def _handle_unprocessable_file(self, remote_file: RemoteFile, format: UnstructuredFormat, logger: logging.Logger) -> None:
-        if format.skip_unprocessable_file_types:
+    def _handle_unprocessable_file(
+        self, remote_file: RemoteFile, skip_unprocessable_file_types: Optional[bool], logger: logging.Logger
+    ) -> None:
```

Comment:
The use of `Optional` is confusing here, especially without a default ` = None` declaration and coming before another parameter that is non-optional. Wdyt of replacing with a simple `bool` type, and resolve upstream to `True` or `False` ahead of calling this method? (We can append `or False` to automatically coalesce `None` to `False` inline, for instance.)